### PR TITLE
Improve ease of safe use of library

### PR DESCRIPTION
**Pull request by thomassa**
Also use those improvements in the CLI tool.

### CODE REVIEW

**Comment by thomassa** [comment]
Note to reviewer: please consider the design of these new functions together with the way they are used in 
https://github.com/xapi-project/xapi-nbd/pull/13 in case offering a different interface here in `nbd` could lead to nicer code in `xapi-nbd`.

**Comment by thomassa** [comment]
I have tried running ocp-indent and it changes many blocks of code, including ones that I've not touched.
Therefore I feel I'd rather do it in a separate pull-request. Maybe I'll do an ocp-indent pullreq first, and rebase these meaningful changes on top if it.

**Comment by thomassa** [comment]
Done: see #84 for the ocp-indent pull-request.
(If that is merged then the first commit in this pull-request should disappear since it is the same.)

**Comment by thomassa** [comment]
Thank you. I'll squash and merge...

File: lib/s.ml [hunk 1]
```diff
@@ -70,4 +70,14 @@ module type SERVER = sig
 
   val close: t -> unit Lwt.t
   (** [close t] shuts down the connection [t] and frees any allocated resources *)
+
+  val with_connection :
+    Channel.cleartext_channel ->
+    ?offer:name list ->
+    (t -> unit Lwt.t) ->
```
Comment (gaborigloi) [inline lib/s.ml]: The type of `f` should be updated to `(string -> t -> 'a Lwt.t)`.

**Pull request merged**